pkg/scheduler/api/validation/validation.go

```diff
@@ -65,6 +73,64 @@ func ValidatePolicy(policy schedulerapi.Policy) error {
 	return utilerrors.NewAggregate(validationErrors)
 }
 
+var FeatureGateDependencies = map[featuregate.Feature]schedulerapi.FeatureDependency{
+	features.TaintNodesByCondition: {
+		NeededPredicateList: sets.NewString(predicates.PodToleratesNodeTaintsPred),
+		NeededPriorityList:  nil,
+		ExcludedPredicateList: sets.NewString(predicates.CheckNodeConditionPred,
+			predicates.CheckNodeMemoryPressurePred,
+			predicates.CheckNodeDiskPressurePred,
+			predicates.CheckNodePIDPressurePred,
+		),
+		ExcludedPriorityList: nil,
+	},
+}
+
+// validateFeatureDependencies checks dependent and excluded predicates/priorities for given feature gates
+func validateFeatureDependencies(policy schedulerapi.Policy) error {
+	var validationErrors []error
+	for feature, featureDependencies := range FeatureGateDependencies {
+		if !utilfeature.DefaultFeatureGate.Enabled(feature) {
+			continue
```

Comment:
When a featuregate is **disabled**, it'd be good to check the **absence** of its dependents. For instance, if the featuregate `TaintNodesByCondition` is disabled, but the predicate is present in the custom policy, it won't work properly, so we should give a validation error.

Additionally, I'm proposing to separate dependents into a) the predicate/priority itself plus b) its (real) dependents. So that we can only check a) instead of a)+b).
It's because we **may** come across some feature which has its own predicate/priority, and also other ones as its dependents. In that case, we shouldn't check the absence of the dependent ones but just its own one.

Thoughts?

Reply:
It would be helpful for the user to check when a feature gate is disabled but its predicate is present, in the case that they forgot to enable the feature, but I think the more important case here is when a feature is enabled and conflicting (excluded) predicates are also enabled. In the first case we are simply reminding the user of something they forgot to enable, which is fine and helpful but not necessarily breaking whereas in the second case an invalid predicate/priority will at least create errors when we remove them in ApplyFeatureGates. Maybe we could check both?

Also, I'm still not sure exactly what you mean in separating the predicate itself plus its dependents. Could you provide an example struct to demonstrate?